modules/apps/object/source-formatter-suppressions.xml

```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 
 <suppressions>
-	<checkstyle>
-		<suppress checks="GenericTypeCheck" files="object-api/src/main/java/com/liferay/object/system/model/listener/SystemObjectDefinitionMetadataModelListener\.java" />
-	</checkstyle>
+	<source-check>
```

Comment:
Why this change?

Reply:
Yep, I prefer to avoid suppressions in SF as long as we can code it according to SF. So, in the future, if someone adds more logic (hope no, it is an `api` module 😁 ) the SF will complain in case they add code with generics that should be checked.
In this case, I made this change
```
@Override
	public void onAfterUpdate(BaseModel originalBaseModel, BaseModel baseModel)
```
To
```
@Override
	public void onAfterUpdate(T originalBaseModel, T baseModel)
```
Note the change of `BaseModel` to T. 
This change applies to more than one method.
Also, I add the generic definition at the top of the class from:
```
public class SystemObjectDefinitionMetadataModelListener
	extends BaseModelListener {
```
to
```
public class SystemObjectDefinitionMetadataModelListener<T extends BaseModel<T>>
	extends BaseModelListener<T> {
```